### PR TITLE
add pubish xyzirt option for liorf

### PR DESCRIPTION
**Pull request by knorth55**
this PR adds publish `XYZIRT` option for `liorf`.
`liorf` only accepts `XYZIRT` format, so I added the option.

### CODE REVIEW

File: livox_ros_driver/livox_ros_driver/lddc.cpp [hunk 1]
```diff
@@ -251,7 +237,111 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
   return published_packet;
 }
 
-void Lddc::FillPointsToPclMsg(PointCloud::Ptr& pcl_msg, \
+void Lddc::FillPointsToPointCloudXYZIRT(PointCloudXYZIRT::Ptr& pcl_msg, \
```
Comment (garaemon) [inline livox_ros_driver/livox_ros_driver/lddc.cpp]: Do we need these back slashes? I understand the original code has them, but ...

File: livox_ros_driver/livox_ros_driver/lddc.cpp [hunk 2]
```diff
@@ -251,7 +237,111 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
...
+    QueuePrePop(queue, &storage_packet);
+    LivoxEthPacket *raw_packet =
+        reinterpret_cast<LivoxEthPacket *>(storage_packet.raw_data);
+    uint32_t single_point_num = storage_packet.point_num * echo_num;
+
+    if (kSourceLvxFile != data_source) {
+      PointConvertHandler pf_point_convert =
+          GetConvertHandler(lidar->raw_data_type);
+      if (pf_point_convert) {
+        pf_point_convert(point_buf, raw_packet, lidar->extrinsic_parameter, \
```
Comment (garaemon) [inline livox_ros_driver/livox_ros_driver/lddc.cpp]: ditto

File: livox_ros_driver/livox_ros_driver/lddc.cpp [hunk 3]
```diff
@@ -251,7 +237,111 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
...
+            line_num);
+      } else {
+        /* Skip the packet */
+        ROS_INFO("Lidar[%d] unkown packet type[%d]", handle,
+                 lidar->raw_data_type);
+        break;
+      }
+    } else {
+      LivoxPointToPxyzrtl(point_buf, raw_packet, lidar->extrinsic_parameter, \
```
Comment (garaemon) [inline livox_ros_driver/livox_ros_driver/lddc.cpp]: ditto

File: livox_ros_driver/livox_ros_driver/lddc.cpp [hunk 4]
```diff
@@ -251,7 +237,111 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
...
+          line_num);
+    }
+
+    LivoxPointXyzrtl *dst_point = (LivoxPointXyzrtl *)point_buf;
+    FillPointsToPointCloudXYZIRT(cloud, dst_point, single_point_num, \
```
Comment (garaemon) [inline livox_ros_driver/livox_ros_driver/lddc.cpp]: ditto

File: livox_ros_driver/livox_ros_driver/lddc.cpp [hunk 5]
```diff
@@ -251,7 +237,111 @@ uint32_t Lddc::PublishPointcloud2(LidarDataQueue *queue, uint32_t packet_num,
...
+      bag_->write(p_publisher->getTopic(), ros::Time(timestamp / 1000000000.0),
+          cloud);
+    }
+  }
+  if (!lidar->data_is_pubulished) {
+    lidar->data_is_pubulished = true;
+  }
+  return published_packet;
+}
+void Lddc::FillPointsToPointCloudXYZI(PointCloudXYZI::Ptr& pcl_msg, \
```
Comment (garaemon) [inline livox_ros_driver/livox_ros_driver/lddc.cpp]: ditto

**Review by garaemon — APPROVED** [review]

File: livox_ros_driver/livox_ros_driver/lddc.cpp (hunk 1 above)
Comment (knorth55) [reply livox_ros_driver/livox_ros_driver/lddc.cpp]: I removed them.

**Pull request merged**